src/bionics_ui.cpp

```diff
@@ -20,15 +20,118 @@ bionic_chars( "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ\"#&()*+./:;@
 
 namespace
 {
-enum bionic_tab_mode {
-    TAB_ACTIVE,
-    TAB_PASSIVE
+using bvector = std::vector <bionic *>;
+
+enum bionic_columns : int {
+    column_status,
+    column_act_cost,
+    column_turn_cost,
+    column_num_entries
 };
-enum bionic_menu_mode {
-    ACTIVATING,
-    EXAMINING,
-    REASSIGNING
+struct bionic_col_data {
+    private:
+        int width;
+        std::string name;
+        std::string description;
+    public:
+        bionic_col_data( int width, std::string name, std::string description ) :
+            width( width ), name( name ), description( description ) {
+        }
+        int get_width() {
+            return std::max( utf8_width( _( name ) ) + 1, width );
+        }
+        std::string get_name() {
+            return _( name );
+        }
+        std::string get_description() {
+            return _( description );
+        }
 };
+bionic_col_data get_col_data( int col_idx )
+{
+    static std::array< bionic_col_data, column_num_entries> col_data = { {
+            {
+                3, translate_marker( "Status" ),
+                string_format( translate_marker( "CBM Status:\n\t<color_%s>+</color> - activated\n\t<color_%s>x</color> - incapacitated. This CBM cannot be used for some time." ),
```

Comment:
That wont work. The translation system can only translate the strings it knows, it can not translate dynamically created strings.

Why are you adding the colors via `string_form_color anyway`? You can just put the color name into the string directly: "<color_green>" and so on.

Reply:
Documentation at output.h 184 says to do that. I guess it's not really clear what color correspond to what string. But seems to work ether way.

Reply:
Have you checked that translation of the actual output works this way?

> Documentation at output.h 184 says to do that.

That applies to dynamically defined colors. Not to fixed color values. The documentation even contains this:
```C++
nc_color color = ... // << this implies it's calculated somehow and not a compile time constant.
```